fix(spec/abci): Added proper description of `ExtendedVoteInfo` and `VoteInfo` (#4460)
cometbft/cometbft: spec/abci/abci++_methods.md

Reviewer: Not sure inf the Filed Number column should follow the proto.

In the addressed case, field 2 became reserved (it was `signed_last_block`), so all other field numbers are incremented by one.

Reviewer: Since we are fixing the documentation, the `VoteInfo` spec also does not match with the protos.

The `signed_last_block` field (2) was replaced by `block_id_flag` field (3) as well.

Author: I aligned it with the protos as this is how it is done in the spec.

    | Name              | Type                    | Description                                                                  | Field Number |
    |-------------------|-------------------------|------------------------------------------------------------------------------|--------------|
    | validator         | [Validator](#validator) | The validator that sent the vote.                                            | 1            |

>     | vote_extension    | bytes                   | Non-deterministic extension provided by the sending validator's Application. | 2            |

Reviewer: ```suggestion
    | vote_extension    | bytes                   | Non-deterministic extension provided by the sending validator's Application. | 3            |
```

>     | extension_signature | bytes             | Vote extension signature created by CometBFT.                | 3            |

Reviewer: ```suggestion
    | extension_signature | bytes             | Signature of the vote extension produced by the sending validator and verified by CometBFT.                | 4            |
```

Author: This was copied from the code comment, the rest is in line with that .

Reviewer: What code? We should then update the code comment, probably, ehehe.

>     | block_id_flag | [BlockIDFlag](../core/data_structures.md#blockidflag)    | block_id_flag indicates whether the validator voted for a block, nil, or did not vote at all                | 4            |

Reviewer 2: ```suggestion
    | block_id_flag | [BlockIDFlag](../core/data_structures.md#blockidflag)    | `block_id_flag` indicates whether the validator voted for a block, nil, or did not vote at all .               | 4            |
```

Reviewer 2: the table formatting looks a bit weird on my IDE but I've checked the rendering an it's fine, so no big deal.

>     | block_id_flag | [BlockIDFlag](../core/data_structures.md#blockidflag)    | `block_id_flag` indicates whether the validator voted for a block, nil, or did not vote at all .               | 4            |

Reviewer: ```suggestion
    | block_id_flag | [BlockIDFlag](../core/data_structures.md#blockidflag)    | Indicates whether the validator voted the last block, nil, or its vote was not received.               | 5            |
```

* **Usage**:
    * Indicates whether a validator signed the last block, allowing for rewards based on validator availability.
    * This information is extracted from CometBFT's data structures in the local process.

>     * `vote_extension` contains the sending validator's vote extension, which is signed by CometBFT. It can be empty

Reviewer: ```suggestion
    * `vote_extension` contains the sending validator's vote extension, whose signature was verified by CometBFT. It can be empty.
```

>     * `extension_signature` contains all the signatures that signed the vote. This way, when this is passed to the application, it can

Reviewer: Why plural? Each entry contains a single extension and single signature.

>     * `extension_signature` contains all the signatures that signed the vote. This way, when this is passed to the application, it can

Reviewer: ```suggestion
    * `extension_signature` is the signature of the vote extension, which was verified verified by CometBFT. This way, when this is passed to the application, it can
```

>     verify the signatures that signed the vote. 

Reviewer: Actually, this is not the point here. Comet has verified the signature, otherwise it would not appear here. We render it available in the case the application needs it for any different reason.
